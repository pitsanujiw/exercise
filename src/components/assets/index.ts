export * from './assetItem';
export * from './assetsTable';
