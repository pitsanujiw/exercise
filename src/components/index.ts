export * from "./assets";
export * from "./loading";
export * from "./orderForm";
