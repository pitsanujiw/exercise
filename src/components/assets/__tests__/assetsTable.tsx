import { AssetTable } from "components";
import { render } from "@testing-library/react";

describe("Asset Table", () => {
  it("Should render it correctly", async () => {
    const renderResult = render(
      <div>
        <AssetTable
          assetList={[
            {
              price: "10",
              amount: "10",
            },
          ]}
          type="sell"
        />
      </div>
    );
    expect(renderResult).toMatchSnapshot();
  });
});
