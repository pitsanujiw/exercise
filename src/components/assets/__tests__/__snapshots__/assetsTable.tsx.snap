// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Table Should render it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root AssetItem-table-5 MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head AssetItem-table-5 MuiTableCell-alignCenter"
                  scope="col"
                >
                  Price (USD)
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head AssetItem-table-5 MuiTableCell-alignCenter"
                  scope="col"
                >
                  Amount (USD)
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root AssetItem-table-5 AssetItem-asset-1 AssetItem-down-3"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
                >
                  $10.00
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
                >
                  $10.00
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiBox-root MuiBox-root-8 AssetItem-title_down-7"
        >
          $10.00
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root AssetItem-table-5 MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head AssetItem-table-5 MuiTableCell-alignCenter"
                scope="col"
              >
                Price (USD)
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head AssetItem-table-5 MuiTableCell-alignCenter"
                scope="col"
              >
                Amount (USD)
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root AssetItem-table-5 AssetItem-asset-1 AssetItem-down-3"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
              >
                $10.00
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body AssetItem-text-4 MuiTableCell-alignCenter"
              >
                $10.00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiBox-root MuiBox-root-8 AssetItem-title_down-7"
      >
        $10.00
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
